### DISCUSSION TITLE
Hexagonal Pore Netwokr

### DISCUSSION THREAD

**Discussion by kunal0kumar0sinha** [PMEAL/OpenPNM#2920] [Q&A]
Can you please help me obtain a pore network like the image attached? 
Thank you 

![hexagon](https://github.com/PMEAL/OpenPNM/assets/123363487/1ce60007-380a-4cbb-8f20-b795c6a94d27)

**Comment by jgostick** [comment]
Your best bet would be to use a Voronoi tessellation. To do this you would generate an image with the `True` voxels representing the junctions, and `False` elsewhere.  Then use `np.where` to extract the `(x,y)` coordinates. You then pass these points into the Voronoi generator class.  You will probably have to identify all connections that are the wrong length and delete them to only keep the ones you want.  Hope that helps.
